primary/src/block_synchronizer/mod.rs

```diff
@@ -40,7 +43,17 @@ pub mod responses;
 /// proceed to next state.
 const CERTIFICATE_RESPONSES_RATIO_THRESHOLD: f32 = 0.5;
 
-type ResultSender<T> = Sender<BlockSynchronizeResult<Certificate<T>>>;
+#[derive(Debug, Clone)]
+pub struct BlockHeader<PublicKey: VerifyingKey> {
+    certificate: Certificate<PublicKey>,
+    /// It designates whether the requested quantity (either the certificate
+    /// or the payload) has been retrieved via the local storage. If true,
+    /// the it used the storage. If false, then it has been fetched via
+    /// the peers.
+    fetched_from_storage: bool,
+}
```

Comment:
I'm still building context on Narwhal but when I see something called a `BlockHeader` it makes me things this is something that would be shared to other peers. Given the actual makeup of this type (ie the presence of the `fetched_from_storage` bit) I take it this is not the case and is intended to remain a local type and never transmitted on the wire, correct?

Reply:
That is correct. I totally see what you mean and when I named it like this I kind of had a similar feeling. I originally wanted to name something like `BlockHeaderResult` but then there would probably be confusion around semantics (e.x this is not a Rust `Result`). I am really open to suggestions.